api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/pac4j/saml/Pac4jSamlClientProperties.java

```diff
@@ -289,6 +289,15 @@ public class Pac4jSamlClientProperties extends Pac4jBaseClientProperties impleme
      */
     private String messageStoreFactory = "org.pac4j.saml.store.EmptyStoreFactory";
 
+    /**
+     * Controls the way SAML attributes are converted from the SAML authentication response into pac4j attributes.
+     * <ul>
+     *     <li>{@code org.pac4j.saml.profile.converter.SimpleSAML2AttributeConverter}: Values of complex types are serialized into a single attribute. (default)</li>
+     *     <li>{@code org.pac4j.saml.profile.converter.ComplexTypeSAML2AttributeConverter}: Values of complex types are unboxed into several attributes; one per child element.</li>
+     * </ul>
+     */
+    private String samlAttributeConverter = "org.pac4j.saml.profile.converter.SimpleSAML2AttributeConverter";
```

Comment:
Does this really need to be the default? What happens if you remove the value here?

Reply:
You are right, the default pac4j configuration will use the SimpleSAML2AttributeConverter anyway, so defining a specific class via the configuration will do the trick if you want to enforce non-default behaviour.